### PR TITLE
Add resource limit type to NotTriggerScaleUp event

### PR DESCRIPTION
**Pull request by aleksandra-malinowska**
Resource limit message was indistinguishable from resource limits message. This change adds the type of resource on which limit would be exceeded.

Had to refactor scale up tests a bit in order to add a unit test without duplicating the entire test setup.

### CODE REVIEW

File: cluster-autoscaler/core/scale_up.go [hunk 1]
```diff
@@ -238,10 +239,14 @@ func (sr *skippedReasons) Reasons() []string {
 
 var (
 	backoffReason         = &skippedReasons{[]string{"in backoff after failed scale-up"}}
-	maxLimitReachedReason = &skippedReasons{[]string{"max limit reached"}}
+	maxLimitReachedReason = &skippedReasons{[]string{"max size reached"}}
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_up.go]: maybe "max node group size reached"? I don't think it's obvious.

File: cluster-autoscaler/core/scale_up.go [hunk 2]
```diff
@@ -238,10 +239,14 @@ func (sr *skippedReasons) Reasons() []string {
...
 	notReadyReason        = &skippedReasons{[]string{"not ready for scale-up"}}
 )
 
+func maxResourceLimitReached(resources []string) *skippedReasons {
+	return &skippedReasons{[]string{fmt.Sprintf("max %s limit reached", strings.Join(resources, ", "))}}
```
Comment (MaciekPytel) [inline cluster-autoscaler/core/scale_up.go]: "cluster %s limit reached"?

File: cluster-autoscaler/core/scale_up.go (hunk 1 above)
Comment (aleksandra-malinowska) [reply cluster-autoscaler/core/scale_up.go]: Done

File: cluster-autoscaler/core/scale_up.go (hunk 2 above)
Comment (aleksandra-malinowska) [reply cluster-autoscaler/core/scale_up.go]: Done

**Comment by MaciekPytel** [comment]
/lgtm
/approve
/hold
@aleksandra-malinowska feel free to cancel hold once you're confident the new messages are more understandable than current ones.

**Pull request merged**